pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -191,6 +191,23 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, updateErr
 	}
 
+	if err := r.checkValuesFileSourcesRefs(ctx, r.client, rs); err != nil {
+		r.logger(ctx).Error(err, "Sync spec invalid")
+		reposync.SetStalled(rs, "Validation", err)
+		// Validation errors should not trigger retry (return error),
+		// unless the status update also fails.
+		_, updateErr := r.updateStatus(ctx, currentRS, rs)
+		// Use the validation error for metric tagging.
+		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
+
+		// If the RSync is being deleted, we still need to run the teardown function,
+		// as we want the user to be able to delete the ConfigMap before the RSync.
+		// Therefore, we only return here if the RSync is not being deleted.
+		if rs.GetDeletionTimestamp().IsZero() {
+			return controllerruntime.Result{}, updateErr
```

Comment:
are there any validation errors that might need to stop progress here that are not NotFound?

Reply:
The errors that we might encounter from this validation check are that we can't fetch the ConfigMap, the ConfigMap isn't immutable, or the referenced ConfigMap doesn't have the specified data key. I don't think any of these errors need to halt teardown or deletion of the RSync.

Reply:
gotcha. so validation errors only block setup, not teardown. makes sense.